agent/tools.py

```
@@ -251,3 +255,23 @@ def stitch_patterns(self, pattern_names: list[str], initial_input: str) -> str:
             current_input = self.apply_pattern(pattern_name, variables)
         
         return current_input
+
+
+    def get_architecture_summary(self) -> dict[str, Any]:
```

Ruff (0.15.12)

[error] Undefined name `Any`

(F821)